node/src/components/block_synchronizer/deploy_acquisition.rs

```diff
@@ -123,12 +119,13 @@ impl<T: Copy + Eq + Hash> Acquisition<T> {
     fn apply_deploy(&mut self, deploy_identifier: T) -> Acceptance {
         if self
             .inner
-            .insert(deploy_identifier, DeployState::HaveDeployBody)
-            .is_none()
+            .contains(&(deploy_identifier, DeployState::HaveDeployBody))
         {
-            Acceptance::NeededIt
-        } else {
             Acceptance::HadIt
+        } else {
+            self.inner
+                .push((deploy_identifier, DeployState::HaveDeployBody));
```

Comment:
Won't it now contain both an entry `(deploy_identifier, DeployState::Vacant)` _and_ `(deploy_identifier, DeployState::HaveDeployBody)`?
And the `HaveDeployBody` entries will be in the order in which we received the deploys, not the order in the block? (Maybe that's fine, but I find it surprising.)

Reply:
it did indeed. I merged before I saw your comment but luckily I independently noticed the issue and fixed it in https://github.com/casper-network/casper-node/pull/3382/files#diff-a3e41acc9e099ca403e69b23f718eff1db0ea277539af686925f24f971a98966R124-R137